### PR TITLE
Improve documentation on how to run tests

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
This is regularly being asked for.

CC @Andersson007 

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
CONTRIBUTING.md

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -36,6 +36,38 @@ If you want to test a PR locally, refer to [our testing guide](https://github.co
...
+
+You have to check out the repository into a specific path structure to be able to run `ansible-test`. The path to the git checkout must end with `.../ansible_collections/community/general`. Please see [our testing guide](https://github.com/ansible/community-docs/blob/main/test_pr_locally_guide.rst) for instructions on how to check out the repository into a correct path structure. The short version of these instructions is:
+
+```.bash
+mkdir -p ~/dev/ansible_collections/community
+git clone https://github.com/ansible-collections/community.general.git ~/dev/ansible_collections/community/general
+cd ~/dev/ansible_collections/community/general
+```
+
+Then you can run `ansible-test` (which is a part of [ansible-core](https://pypi.org/project/ansible-core/)) inside the checkout. The following example commands expect that you have installed Docker or Podman. Note that Podman has only been supported by more recent ansible-core releases. If you are using Docker, the following will work with other
```
Comment (Andersson007) [inline CONTRIBUTING.md:49]:
```suggestion
Then you can run `ansible-test` (which is a part of [ansible-core](https://pypi.org/project/ansible-core/)) inside the checkout. The following example commands expect that you have installed Docker or Podman. Note that Podman has only been supported by more recent ansible-core releases. If you are using Docker, the following will work with other:
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -36,6 +36,38 @@ If you want to test a PR locally, refer to [our testing guide](https://github.co
...
+```.bash
+mkdir -p ~/dev/ansible_collections/community
+git clone https://github.com/ansible-collections/community.general.git ~/dev/ansible_collections/community/general
+cd ~/dev/ansible_collections/community/general
+```
+
+Then you can run `ansible-test` (which is a part of [ansible-core](https://pypi.org/project/ansible-core/)) inside the checkout. The following example commands expect that you have installed Docker or Podman. Note that Podman has only been supported by more recent ansible-core releases. If you are using Docker, the following will work with other 
+
+```.bash
+# Run sanity tests:
```
Comment (Andersson007) [inline CONTRIBUTING.md:52]:
```suggestion
# Run sanity tests for all files in the collection (alternatively you can run tests against a certain fail by passing its path to the command):
```
Or something like that (or maybe something like you wrote for unit tests? i.e. couple of separate examples)

**Review by Andersson007 — COMMENTED** [review]
@felixfontein thanks for the PR! Couple of minor things.
Also there's a typo in line 27 `respository` - while you're here.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -36,6 +36,38 @@ If you want to test a PR locally, refer to [our testing guide](https://github.co
...
+
+# Run all unit tests:
+ansible-test units --docker -v
+# Run all unit tests for one Python version (a lot faster):
+ansible-test units --docker -v --python 3.8
+# Run a specific unit test (for the nmcli module) for one Python version:
+ansible-test units --docker -v --python 3.8 tests/unit/plugins/modules/net_tools/test_nmcli.py
+
+# Run integration tests for the interfaces_files module in a Docker container using the
+# fedora35 operating system image (the supported images depend on your ansible-core version)
```
Comment (Andersson007) [inline CONTRIBUTING.md:63]:
```suggestion
# fedora35 operating system image (the supported images depend on your ansible-core version):
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -36,6 +36,38 @@ If you want to test a PR locally, refer to [our testing guide](https://github.co
...
+ansible-test integration --docker fedora35 -v interfaces_file
+# Run integration tests for the flattened lookup **without any isolation**
```
Comment (Andersson007) [inline CONTRIBUTING.md:65]:
```suggestion
# Run integration tests for the flattened lookup **without any isolation**:
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -36,6 +36,38 @@ If you want to test a PR locally, refer to [our testing guide](https://github.co
...
+
+Then you can run `ansible-test` (which is a part of [ansible-core](https://pypi.org/project/ansible-core/)) inside the checkout. The following example commands expect that you have installed Docker or Podman. Note that Podman has only been supported by more recent ansible-core releases. If you are using Docker, the following will work with other 
+
+```.bash
+# Run sanity tests:
+ansible-test sanity --docker -v
+
+# Run all unit tests:
+ansible-test units --docker -v
+# Run all unit tests for one Python version (a lot faster):
```
Comment (Andersson007) [inline CONTRIBUTING.md:57]:
```suggestion

# Run all unit tests for one Python version (a lot faster):
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -36,6 +36,38 @@ If you want to test a PR locally, refer to [our testing guide](https://github.co
...
+ansible-test units --docker -v --python 3.8
+# Run a specific unit test (for the nmcli module) for one Python version:
```
Comment (Andersson007) [inline CONTRIBUTING.md:59]:
```suggestion

# Run a specific unit test (for the nmcli module) for one Python version:
```

File: CONTRIBUTING.md (hunk 4 above)
Comment (Andersson007) [inline CONTRIBUTING.md:65]:
```suggestion

# Run integration tests for the flattened lookup **without any isolation**
```

File: CONTRIBUTING.md (hunk 6 above)
Comment (Andersson007) [reply CONTRIBUTING.md:59]: for readability

File: CONTRIBUTING.md (hunk 1 above)
Comment (felixfontein) [reply CONTRIBUTING.md]: Sorry, I forgot to complete that sentence. I've pushed 6702ef235b83569c310e871de844df1cdf4b26ff to fix that.

**Comment by felixfontein** [comment]
> Also there's a typo in line 27 `respository` - while you're here.

Thanks, fixed that!

**Review by Andersson007 — APPROVED** [review]
LGTM, thanks!

**Comment by felixfontein** [comment]
@Andersson007 thanks for reviewing this!

**Pull request merged**